### PR TITLE
[DO NOT MERGE] update ec2 infrastructure with minimal (single node) and multi_node versions

### PR DESCRIPTION
**Pull request by johnkim76**

### CODE REVIEW

File: ansible/roles/aws_repo_setup/tasks/main.yml [hunk 1]
```diff
@@ -1,16 +1,52 @@
 ---
-  - name: install EPEL
```
Comment (jwmatthews) [inline ansible/roles/aws_repo_setup/tasks/main.yml]: If I'm reading this right, you removed EPEL.

You don't want to remove it, you want to support multiple ways of using content:
 - EPEL
 - RedHat CDN
 - Mirror Repos

Might be each is written in it's own task file, then you include the one you want based on some variables.

File: config/my_vars.yml.example [hunk 2]
```diff
@@ -8,6 +8,8 @@ dockerhub_org_name: example_org
 # Omit this unless you want to store your password in plain text to skip prompts.
 # dockerhub_user_password: example_password
 
+#demo: False
```
Comment (jwmatthews) [inline config/my_vars.yml.example]: I think we can remove this, don't think we want "demo" setting in my_vars

**Comment by jwmatthews** [comment]
First, it's nice to see this coming together, will be a big help being able to kick off multi-node clusters with a single script.

Let's add a README in ec2 directory specifying the difference between minimal/multi-node

ec2/multi-node needs a README explaining what it is doing and how to invoke it, explain how it uses openshift-ansible to launch a mutli-node cluster, will provision needed infrastructure and run setup.  Note any requirements like the route53 zone.   Add a note up front this is for a multi-node cluster, if you want a simpler single node instance to try out you should look at ec2/minimal which uses "oc cluster up"


ec2/minimal/README should be updated to clearly call out this sets up a single instance and uses "oc cluster up", then add a note if you want to run "openshift-ansible" to deploy a multi-node cluster see "ec2/multi-node"

In the main README of catasb, call out that for the ec2 deployments we support:
 - minimal setup that provisions a single instance and leverages "oc cluster up --service-catalog"
 - multi-node setup that provisions multiple instances and leverages "openshift-ansible"


We need to support/test multiple flows of where content comes from (and document how to configure each):
 - Content comes from RHEL repos from CDN, meaning we need to use subscription manager to register
 - Content comes from a mirror repo to test latest puddles
  (Content coming from CentOS repos would be interesting as well, but might be too much for now)


I have concern with the new role 'aws_infrastructure' and the existing role 'infrastructure', feels like there will be confusion in future of which one should be used.

'infrastructure' was originally used to create the VPC and associated infrastructure required in the account plus the instance for the single node case in EC2.

I would expect we'd look to merge the roles and allow a single role to satisfy needs of both:
 - minimal 
 - multi-node

The name of 'aws_infrastructure" for the role is good, see what it would look like to merge 'infrastructure' role into 'aws_infrastructure'.

Then be sure to test minimal and as well as multi-node invocations.

File: config/my_vars.yml.example (hunk 2 above)
Comment (johnkim76) [inline config/my_vars.yml.example]: thank you. will update

File: ansible/roles/aws_repo_setup/tasks/main.yml (hunk 1 above)
Comment (johnkim76) [inline ansible/roles/aws_repo_setup/tasks/main.yml]: thank you.. will update as suggested

**Comment by johnkim76** [comment]
thank you.. I will update the README's and combine the infrastructure as suggested.
I will also test the repo and other config changes you mentioned.

**Comment by johnkim76** [comment]
[(13) Automate Stand-Up of Openshift Cluster to AWS](https://trello.com/c/FkuHzqHM/404-13-automate-stand-up-of-openshift-cluster-to-aws)

File: ec2/multi_node/README.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      File "<string>", line 1, in <module>
+      ImportError: No module named ansible
+
+      brew uninstall ansible
+      pip install ansible
+
+      $ python -c "import ansible;print(ansible.__version__)"
+      2.3.0.0
+      ```
+  * Install python dependencies (This is needed for python2. Use pip2 if using python3)
```
Comment (jmontleon) [inline ec2/multi_node/README.md]: pip3 if using python3?

File: ec2/multi_node/README.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    * On Fedora and EL7 it is recommended that you use ansible in a python virtualenv.
+     * This is due to a couple reasons:
+       - boto rpms are not sufficiently new enough
+       - pip is not sudo safe on Fedora and EL7 
+     * To setup and active a virtualenv do the following;
```
Comment (jmontleon) [inline ec2/multi_node/README.md]: activate. Sorry. I know I wrote this... I apparently did not have enough coffee when I did it...

**Comment by jmontleon** [comment]
@johnkim76 Nice! I have not run it yet, but it looks good. There are a couple typos that were my fault in the README.md that I mentioned which can be fixed.

File: ec2/multi_node/README.md (hunk 4 above)
Comment (johnkim76) [inline ec2/multi_node/README.md]: Sorry, I just copied this part over.. I will change it

**Comment by johnkim76** [comment]
this PR will be closed and a NEW PR will be opened with a fusor/catasb branch

**Comment by johnkim76** [comment]
new PR created.. closing this one
https://github.com/fusor/catasb/pull/126

**Pull request closed without merging**